pkg/trace/writer/trace.go

```diff
@@ -30,6 +30,18 @@ const pathTraces = "/api/v0.2/traces"
 // a flush is triggered; replaced in tests.
 var MaxPayloadSize = 3200000 // 3.2MB is the maximum allowed by the Datadog API
 
+type prioritySampler interface {
+	GetTargetTPS() float64
+}
+
+type errorsSampler interface {
+	GetTargetTPS() float64
+}
```

Comment:
Is there a reason the interface is tied to the samplers and not to the functionality? For example the interface here could be `TargetTPSReader` that has the `GetTargetTPS()` method. Similar to `Reader/Writer` methods. If we plan on adding specific methods to each sampler type then it would make sense to keep the separate interfaces.

Reply:
Fully agree. I tried to mimick https://github.com/DataDog/datadog-agent/blob/512c623002b66d23b26bde3846da8ddb8d25a75a/pkg/trace/remoteconfighandler/remote_config_handler.go#L16 but what you suggested is better. Will change and try something for the rare sampler snowflake

Reply:
Addressed, I tweaked a bit the name since the function signature was a bit too cryptic IMO. Happy to revise that if you have a better idea.

Reply:
The revised version looks good.